plugins/aea-ledger-solana/aea_ledger_solana/crypto.py

```diff
@@ -171,14 +172,22 @@ def _sign_tx_legacy(
         :param signers: list of signers
         :return: signed transaction
         """
-        signers = signers or []
-        jsonTx = json.dumps(transaction)
 
-        keypair = Keypair.from_base58_string(self.private_key)
+        signers = signers or []
         signers = [Keypair.from_base58_string(signer.private_key) for signer in signers]
-        signers.append(keypair)
+        signers.append(self.entity)
+
         recent_hash = Hash.from_string(transaction["recentBlockhash"])
-        stxn = Transaction.from_json(jsonTx)
+        if isinstance(transaction["message"], list):
+            transaction["message"][1]["recentBlockhash"] = recent_hash
+            return json.loads(
+                VersionedTransaction(
+                    message=MessageV0.from_json(json.dumps(transaction["message"][1])),
+                    keypairs=signers,
+                ).to_json()
+            )
```

Comment:
@Dassy23 Please check this logic for signing versioned txs, I ported this from the issue you opened